octobot_trading/personal_data/exchange_personal_data.py

```diff
@@ -298,13 +298,20 @@ async def handle_closed_order_update(self, exchange_order_id, raw_order) -> bool
     async def check_and_update_inactive_orders_when_necessary(
         self, symbol: str, current_price: decimal.Decimal, price_time: float,
         strategy_timeout: typing.Optional[float], wait_for_fill_callback: typing.Optional[typing.Callable]
-    ):
-        for order in self.orders_manager.get_all_orders(symbol=symbol, active=False):
+    ) -> int:
+        activated_orders_count = 0
+        sorted_inactive_orders = sorted(
+            self.orders_manager.get_all_orders(symbol=symbol, active=False),
+            key= lambda o: o.origin_price if o.trigger_above else -o.origin_price
+        )
+        for order in sorted_inactive_orders:
             if order.should_become_active(price_time, current_price):
+                activated_orders_count += 1
```

Comment:
shouldn't it be after the `on_active_trigger` call?

Reply:
Even if the next line fails, we want to know that something might have happened. I'll rename this variable, it’s not really how many got activated but more how many got changed

Reply:
understood